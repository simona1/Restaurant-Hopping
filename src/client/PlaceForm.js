import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng,
} from 'react-places-autocomplete';

import React from 'react';
import {connect} from 'react-redux';

import GOOGLE_API_KEY from '../../api_key';

import updateLocationQuery from './actions/updateLocationQuery';

class PlaceForm extends React.Component<Props> {
  render() {
    return (
      <form className="input-field">
        <i className="material-icons prefix">search</i>
        <input
          type="text"
          id="place"
          placeholder="search for establishments nearby"
          value={this.props.locationQuery}
          onChange={this._handleLocationQueryChange}
        />
      </form>
    );
  }

  _handleLocationQueryChange = event => {
    this.props.onLocationQueryChange(event.target.value);
  };
}

function mapStateToProps(state) {
  return {
    locationQuery: state.locationQuery,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onLocationQueryChange: locationQuery =>
      dispatch(updateLocationQuery(locationQuery)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(PlaceForm);
