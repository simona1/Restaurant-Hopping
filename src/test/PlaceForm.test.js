/*global google*/
import React from 'react';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import PlaceForm from '../client/PlaceForm';
import {Provider} from 'redux';

//const store = {};

describe('Renders default PlaceForm', () => {
  it('should be defined', () => {
    expect(PlaceForm).toBeDefined();
  });
  xit('should render without crashing', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <PlaceForm />
      </Provider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  xit('should contain nested nodes', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <PlaceForm />
      </Provider>,
    );
    expect(wrapper.contains('#place'));
    expect(wrapper.contains('form'));
  });
});
