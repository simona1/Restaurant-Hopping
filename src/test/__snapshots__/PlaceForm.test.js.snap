// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders default PlaceForm should render without crashing 1`] = `<Connect(PlaceForm) />`;
